server_tests/test/clientTests.js

```diff
@@ -791,514 +736,468 @@ describe("RUST API TEST SUITE", function() {
-            .then( Comparison.catchExceptions(done, function(res) {
-                Comparison.invalidParameterValueError(res);
-                done();
-            }));
+        it("Expect invalidParameterValue Error. camera.setOptions cannot set options when wifiPassword option is set to unsupported value", function() {
+            return testClient.setOptions(sessionId, {'wifiPassword': '1234'})
+            .then( shouldNotGetHere, function onError (err) {
+                Comparison.invalidParameterValueError(err);
+            });
         });
 
-        it("Expect missingParameter Error. camera.setOptions cannot set options when options is not provided", function(done) {
-            testClient.setOptions(sessionId, undefined)
-            .then( Comparison.catchExceptions(done, function(res) {
-                Comparison.missingParameterError(res);
-                done();
-            }));
+        it("Expect missingParameter Error. camera.setOptions cannot set options when options is not provided", function() {
+            return testClient.setOptions(sessionId, undefined)
+            .then( shouldNotGetHere, function onError (err) {
+                Comparison.missingParameterError(err);
+            });
         });
     });
 
     // OSC COMMAND STATUS
     describe("Testing /osc/commands/status endpoint", function() {
         var sessionId;
 
-        before( function(done) {
-            testClient.startSession()
-            .then( Comparison.catchExceptions(done, function(res) {
+        before( function() {
+            return testClient.startSession()
+            .then( function onSuccess (res) {
                 sessionId = res.body.results.sessionId;
                 Comparison.oscSessionOpOutput(res, {'sessionId': sessionId});
-                done();
-            }));
+            }, shouldNotGetHere);
         });
 
-        after( function(done) {
-            testClient.closeSession(sessionId)
-            .then( Comparison.catchExceptions(done, function(res) {
+        after( function() {
+            return testClient.closeSession(sessionId)
+            .then( function onSuccess (res) {
                 Comparison.oscCloseSessionOutput(res);
-                done();
-            }));
+            }, shouldNotGetHere);
         });
 
-        it("Expect success. /osc/commands/status successfully grabs command status after take picture has been called", function(done) {
+        it("Expect success. /osc/commands/status successfully grabs command status after take picture has been called", function() {
             this.timeout(timeoutValue);
-            testClient.takePicture(sessionId, Comparison.catchExceptions(done, function(res) {
-                 Comparison.oscCommandsStatusOutput(res, {'name': 'camera.takePicture', 'id': res.body.id});
-            }))
-            .then( Comparison.catchExceptions(done, function(res) {
+            return testClient.takePicture(sessionId, function onSuccess (res) {
```

Comment:
This name doesn't seem right, this is the status callback, not a success or completion callback.